packages/protocol/contracts/layer1/based/ForcedInclusionStore.sol

```diff
@@ -4,91 +4,92 @@ pragma solidity ^0.8.24;
-    }
-
-    function getRequiredPriorityFee() public view returns (uint256) {
-        uint256 queueLength = tail - head;
-        if (queueLength == 0) {
-            return basePriorityFee;
-        }
-
-        return (2 ** queueLength).max(4096) * basePriorityFee;
-    }
-    function storeForcedInclusion(bytes32 blobHash, uint32 blobByteOffset, uint32 blobByteSize) payable external {
-        uint256 requiredPriorityFee = getRequiredPriorityFee();
-        require(msg.value == requiredPriorityFee, ForcedInclusionInsufficientPriorityFee());
-
-        uint64 id = tail + 1;
-        ForcedInclusion memory forcedInclusion = ForcedInclusion({
+    function storeForcedInclusion(
+        uint8 blobId,
+        uint32 blobByteOffset,
+        uint32 blobByteSize
+    )
+        external
+        payable
+    {
+        bytes32 blobHash = _blobHash(blobId);
+        require(blobHash != bytes32(0), BlobNotFound());
+        require(msg.value == fee, IncorrectFee());
+
+        ForcedInclusion memory inclusion = ForcedInclusion({
             blobHash: blobHash,
+            fee: msg.value,
+            createdAt: uint64(block.timestamp),
             blobByteOffset: blobByteOffset,
-            blobByteSize: blobByteSize,
-            timestamp: block.timestamp,
-            priorityFee: msg.value,
-            id: id,
-            processed: false
+            blobByteSize: blobByteSize
         });
 
-        forcedInclusionQueue[tail] = forcedInclusion;
-
-        tail++;
+        queue[tail++] = inclusion;
 
-        emit ForcedInclusionStored(forcedInclusion);
+        emit ForcedInclusionStored(inclusion);
     }
 
-     function consumeForcedInclusion() external override returns (ForcedInclusion memory) {
-        address operator = resolve(LibStrings.B_TAIKO_FORCED_INCLUSION_INBOX, false);
-        require(msg.sender == operator, NotTaikoForcedInclusionInbox());
-
-        ForcedInclusion memory forcedInclusion;
-        if (head == tail) {
-            return forcedInclusion;
-        }
-        
+    function consumeForcedInclusion(address _feeRecipient)
+        external
+        onlyFromNamed(LibStrings.B_TAIKO_FORCED_INCLUSION_INBOX)
+        returns (ForcedInclusion memory inclusion_)
+    {
         // we only need to check the first one, since it will be the oldest.
-        ForcedInclusion storage inclusion = forcedInclusionQueue[head];
-        if (inclusion.timestamp + inclusionWindow < block.timestamp) {
-            inclusion.processed = true;
-            ForcedInclusion memory consumedInclusion = forcedInclusionQueue[head];
-            head++;
+        ForcedInclusion storage inclusion = queue[head];
 
-            emit ForcedInclusionConsumed(consumedInclusion);
-            
-            return consumedInclusion;
+        if (inclusion.createdAt != 0 && block.timestamp >= inclusionDelay + inclusion.createdAt) {
+            inclusion_ = inclusion;
+            _feeRecipient.sendEtherAndVerify(inclusion.fee);
```

Comment:
you're sending the fee twice now, because you already send it outside of the store when it's consumed

Reply:
fixed